src/main/java/uk/gov/companieshouse/pscfiling/api/service/FilingDataServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+package uk.gov.companieshouse.pscfiling.api.service;
+
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.model.filinggenerator.FilingApi;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.pscfiling.api.exception.FilingResourceNotFoundException;
+import uk.gov.companieshouse.pscfiling.api.mapper.PscIndividualMapper;
+import uk.gov.companieshouse.pscfiling.api.model.PscTypeConstants;
+import uk.gov.companieshouse.pscfiling.api.model.entity.Date3Tuple;
+import uk.gov.companieshouse.pscfiling.api.model.entity.NameElements;
+import uk.gov.companieshouse.pscfiling.api.model.entity.PscIndividualFiling;
+import uk.gov.companieshouse.pscfiling.api.utils.LogHelper;
+import uk.gov.companieshouse.pscfiling.api.utils.MapHelper;
+
+/**
+ * Produces Filing Data format for consumption as JSON by filing-resource-handler external service.
+ */
+@Service
+public class FilingDataServiceImpl implements FilingDataService {
+
+    private final PscFilingService pscFilingService;
+    private final PscIndividualMapper pscIndividualMapper;
+    private final PscDetailsService pscDetailsService;
+    private final Logger logger;
+
+    public FilingDataServiceImpl(PscFilingService pscFilingService,
+                                 PscIndividualMapper filingMapper,
+                                 PscDetailsService pscDetailsService,
+                                 Logger logger) {
+        this.pscFilingService = pscFilingService;
+        this.pscIndividualMapper = filingMapper;
+        this.pscDetailsService = pscDetailsService;
+        this.logger = logger;
+    }
+
+    @Override
+    public FilingApi generatePscFiling(String filingId, Transaction transaction, String passthroughHeader) {
+        var filing = new FilingApi();
+        filing.setKind("psc-filing#cessation"); // TODO: handling other kinds to come later
+
+        setFilingApiData(filing, filingId, transaction, passthroughHeader);
+        return filing;
+    }
+
+    private void setFilingApiData(FilingApi filing, String filingId, Transaction transaction, String passthroughHeader) {
```

Comment:
Suggestion: since this method modifies the `filing` argument, it's usual to return it from the method, and could rename so the usage becomes:
```
filing = populateFilingData(filing, filingId, transaction, passthroughHeader);
```